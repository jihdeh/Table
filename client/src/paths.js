export default {
  startScreen: '/',
  play: '/play'
}
