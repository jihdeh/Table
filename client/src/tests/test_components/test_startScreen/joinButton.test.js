import JoinButton from '../../../components/startScreen/joinButton';
import paths from '../../../paths';

import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();

it('component renders as before', () => {
  const component = shallowRenderer.render(<JoinButton />);
  expect(component).toMatchSnapshot();
});

jest.mock('../../../helpers/routing', () => ({
  goToPath: jest.fn()
}));

it('goes to play url on join button click', () => {
  const component = shallow(<JoinButton />);
  component.simulate('click');
  const goToPath = require.requireMock('../../../helpers/routing').goToPath;
  expect(goToPath).toHaveBeenCalledWith(paths.play);
});
