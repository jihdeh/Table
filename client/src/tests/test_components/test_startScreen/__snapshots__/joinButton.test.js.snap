exports[`test component renders as before 1`] = `
<button
  className="player-join-button"
  onClick={[Function]}>
  Join as player
</button>
`;
